src/Components/VideoCard/VideoCard.jsx

```diff
@@ -31,6 +49,116 @@ function VideoCard({ video })
+            else
+            {
+                if(isVideoPresentInWatchLater)
+                {
+                    // Item already present in watch later list
+                    showToast("success","","Video already present in Watch Later")
+                }
+                else
+                {
+                    // Item not present in Watch Later
+                    // Update in backend and then in frontend
+                    let updatedUserInfo = await axios.patch(
+                        "https://videoztron.herokuapp.com/api/watchlater",
+                        {
+                            video
+                        },
+                        {
+                            headers:
+                            {
+                                'x-access-token': localStorage.getItem('token'),
+                            }
+                        }
+                    )
+
+                    if(updatedUserInfo.data.status==="ok")
+                    {
+                        dispatchWatchLaterList({type: "UPDATE_WATCH_LATER_LIST",payload: updatedUserInfo.data.user.watchLater})
+                        setIsVideoPresentInWatchLater(true)
+                        showToast("success","","Video successfully added to watch later")
+                    }
+                }
+            }
+        }
+        else
+        {
+            showToast("warning","","Kindly Login")
+        }
+    }
+
+    const removeItemFromWatchLaterList = async () => {
+        const token=localStorage.getItem('token')
+
+        if(token)
+        {
+            const user = jwt_decode(token)
+                
+            if(!user)
+            {
+                localStorage.removeItem('token')
+                showToast("warning","","Kindly Login")
+                navigate('/login')
+            }
+            else
+            {
+                // Item already present in watch later list, remove it
+                let updatedUserInfo = await axios.delete(
+                    `https://videoztron.herokuapp.com/api/watchlater/${video._id}`,
+                    {
+                        headers:
+                        {
+                            'x-access-token': localStorage.getItem('token'),
+                        }
+                    },
+                    {
+                        video
+                    }
+                )
+                    
+                if(updatedUserInfo.data.status==="ok")
+                {
+                    dispatchWatchLaterList({type: "UPDATE_WATCH_LATER_LIST",payload: updatedUserInfo.data.user.watchLater})
+                    setIsVideoPresentInWatchLater(false)
+                    showToast("success","","Video successfully removed to watch later")
+                }
+            }
+        }
+        else
+        {
+            showToast("warning","","Kindly Login")
+        }
+    }
 
     return (
         <Link to={`/video/${_id}`} state={{videoDetails:video}}>
```

Comment:
Didn't know we can use state with Link like this, learned something new

Reply:
Cool 😄